adapter/internal/messaging/revoked_token_listener.go

```diff
@@ -31,48 +31,47 @@ import (
 func handleTokenRevocation() {
 	for d := range msg.RevokedTokenChannel {
 		var notification msg.EventTokenRevocationNotification
-		unmarshalErr := json.Unmarshal([]byte(string(d.Body)), &notification)
+		unmarshalErr := parseRevokedTokenJSONEvent([]byte(string(d.Body)), &notification)
 		if unmarshalErr != nil {
-			logger.LoggerInternalMsg.Errorf("Error occurred while unmarshalling revoked token event data %v", unmarshalErr)
 			continue
 		}
 		logger.LoggerInternalMsg.Infof("Event %s is received", notification.Event.PayloadData.Type)
 		logger.LoggerInternalMsg.Printf("RevokedToken: %s, Token Type: %s", notification.Event.PayloadData.RevokedToken,
 			notification.Event.PayloadData.Type)
-		var stokens []types.Resource
-		t := &keymgt.RevokedToken{}
-		t.Jti = notification.Event.PayloadData.RevokedToken
-		t.Expirytime = notification.Event.PayloadData.ExpiryTime
-		stokens = append(stokens, t)
-		xds.UpdateEnforcerRevokedTokens(stokens)
+		processTokenRevocationEvent(&notification)
 		d.Ack(false)
 	}
 	logger.LoggerInternalMsg.Infof("handle: deliveries channel closed")
 }
 
 func handleAzureTokenRevocation() {
 	for d := range msg.AzureRevokedTokenChannel {
-		logger.LoggerInternalMsg.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] message received for " +
-			"RevokedTokenChannel = " + string(d))
 		var notification msg.EventTokenRevocationNotification
 		error := parseRevokedTokenJSONEvent(d, &notification)
 		if error != nil {
-			logger.LoggerInternalMsg.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error while processing " +
-				"the token revocation event %v. Hence dropping the event", error)
 			continue
 		}
-		logger.LoggerInternalMsg.Infof("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Event %s is received",
-			notification.Event.PayloadData.Type)
-		logger.LoggerInternalMsg.Printf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] RevokedToken: %s, " +
-			"Token Type: %s", notification.Event.PayloadData.RevokedToken,
+		logger.LoggerInternalMsg.Infof("Event %s is received", notification.Event.PayloadData.Type)
+		logger.LoggerInternalMsg.Printf("RevokedToken: %s, Token Type: %s", notification.Event.PayloadData.RevokedToken,
 			notification.Event.PayloadData.Type)
+		processTokenRevocationEvent(&notification)
 	}
 }
 
+func processTokenRevocationEvent(notification *msg.EventTokenRevocationNotification)  {
+	var stokens []types.Resource
+	t := &keymgt.RevokedToken{}
```

Comment:
shall we  use better variable names for t and stokens

Reply:
Done